pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -403,28 +403,43 @@ public int sendSegmentJson(URI uri, String jsonString, @Nullable List<Header> he
    *
    * @param uri URI
    * @param jsonString Segment json string
-   * @return Response status code
-   * @throws Exception
+   * @return Response
+   * @throws IOException
+   * @throws HttpErrorStatusException
    */
-  public int sendSegmentJson(URI uri, String jsonString) throws Exception {
+  public SimpleHttpResponse sendSegmentJson(URI uri, String jsonString) throws IOException, HttpErrorStatusException {
     return sendSegmentJson(uri, jsonString, null, null, DEFAULT_SOCKET_TIMEOUT_MS);
   }
 
+  /**
+   * Send segment completion protocol request.
+   *
+   * @param uri URI
+   * @param socketTimeoutMs Socket timeout in milliseconds
+   * @return Response
+   * @throws IOException
+   * @throws HttpErrorStatusException
+   */
+  public SimpleHttpResponse sendSegmentCompletionProtocolRequest(URI uri, int socketTimeoutMs) throws Exception {
```

Comment:
`IOException, HttpErrorStatusException` instead of `Exception`

Reply:
Wow, good catch!